### PR TITLE
Fixes the multipage issue #3: the URL of the form was not well updated.

### PR DESCRIPTION
**Pull request by giliam**
For now, I just added a small fix to make sure "displayed_by_question" forms are working. Work asked for in issue will require a lot of additional work.

### CODE REVIEW

File: survey/templates/survey/survey.html [hunk 1]
```diff
@@ -11,7 +11,12 @@
 {% block body %}
 	<h1> {{survey.name}} </h1>
 	<div class="survey-description"> {{survey.description|safe}} </div>
-	<div><form action="{% url 'survey-detail' id=survey.id %}" method="post">
+	<div>
+	{% if step != 0 and step != "" and step is not None %}
```
Comment (Pierre-Sassoulas) [inline survey/templates/survey/survey.html]: Would this work?
```suggestion
      {% if step and step != 0 %}
```

File: survey/views/survey_detail.py [hunk 2]
```diff
@@ -60,14 +64,23 @@ def post(self, request, *args, **kwargs):
             next_url = form.next_step_url()
             response = None
             if survey.display_by_question:
+                # when it's the last step
                 if not form.has_next_step():
                     save_form = ResponseForm(
                         request.session[session_key], survey=survey, user=request.user
                     )
-                    response = save_form.save()
+                    # checks that the form created is valid
```
Comment (Pierre-Sassoulas) [inline survey/views/survey_detail.py]: We can probably remove this comment

File: survey/templates/survey/survey.html (hunk 1 above)
Comment (giliam) [reply survey/templates/survey/survey.html]: Oh, obviously yes, sorry!

**Comment by giliam** [comment]
> This seems straightforward, thank you for fixing this, it has been broken for a long time (since before I took the project and ported it to python3).

By the way, I wanted to have, if possible, a little bit of background around the project. It seems like there are two layers of code, a kind of chaotic one wrapped in quite high quality development processes (testing, precommit, code coverage, etc. Is it possible to have a general explanation of what have happened, what is going to happen, who still works on this, what are the development goals, who still uses this?
I am a little bit unsettled: the core code seems rather unstable but quite substantial. Debugging it (especially the multiple answers system) could be dull and complex. At the same time, starting a new app for surveys seems stupid.

**Comment by giliam** [comment]
I'm just going to integrate some tests, if you don't mind!

**Comment by Pierre-Sassoulas** [comment]
> I wanted to have, if possible, a little bit of background around the project.

I followed the same principle than you right now when I had to host a survey in 2016. I did not want to start from scratch so I took the most popular python django survey's app (that happened to be jessykate's at the time). I added tests, coverages, refactored a lot in order to be able to be confident about the stability of my surveys (100+ questions,  and 1000+ answerers). I also ported it to python3 (while keeping the compatibility with python2.7 until very recently). I added what I needed for my survey while keeping it possible for old project to be able to migrate (thus the database design still looks relatively close to what was done originally, because migrations are hard to do properly). Also, added a LaTeX report generation and csv export from the results that I needed and that was'nt super hard to make generics.

I had no use of the project since I delivered the report at the end of 2017, so some thing that could be considered basic (multiple page survey) are not done, because the state of the code, while not ideal, was sufficient to what I wanted to do at the time. . Since then, the project got some visibility and issues and merge request are opened somewhat regularly by poeple using it. In particular translations in multiple languages were done and merged. I think poeple using the project care more about translation than clean code :)

I did some quotes for freelance work, but was never paid, so right now maintening the project is only voluntary work I do in my free time. I'm not planning on adding anything significant to the project, but I'm going to be accepting good merge requests as long as I'm motivated.

**Comment by Pierre-Sassoulas** [comment]
I think we can break compatibility if you want to change the database design significantly without bothering with the migrations. This is probably a constraint I could have released.

**Comment by giliam** [comment]
Thanks for the honest response, it seems much clearer to me now! I don't have any opinion on the compatibility, so we should stay like that for now maybe (or prepare a 2.0 version maybe in case of compatibility failure?).
Are there still things to do on this PR?

File: survey/tests/testdump.json [hunk 3]
```diff
@@ -94,14 +94,26 @@
  "model": "survey.survey",
  "pk": 4,
  "fields": {
-  "name": "Test surv\u00eby2",
```
Comment (Pierre-Sassoulas) [inline survey/tests/testdump.json]: Actually, the Unicode in the name is part of the test :)

File: survey/views/survey_detail.py [hunk 4]
```diff
@@ -60,20 +64,28 @@ def post(self, request, *args, **kwargs):
...
                 response = form.save()
 
+            # if there is a next step
             if next_url is not None:
                 return redirect(next_url)
             else:
                 del request.session[session_key]
                 if response is None:
-                    return redirect("/")
+                    return redirect(reverse("survey-list"))
```
Comment (Pierre-Sassoulas) [inline survey/views/survey_detail.py]: Nice catch.

File: survey/tests/views/test_survey_detail.py [hunk 5]
```diff
@@ -6,7 +6,7 @@
 from django.urls.base import reverse
 
 from survey.tests import BaseTest
-from survey.models import Response
+from survey.models import Response, Survey, Answer
```
Comment (Pierre-Sassoulas) [inline survey/tests/views/test_survey_detail.py]:
```suggestion
from survey.models import Response, Answer
```

File: survey/tests/testdump.json [hunk 6]
```diff
@@ -94,14 +94,26 @@
...
+  "name": "Test",
```
Comment (Pierre-Sassoulas) [inline survey/tests/testdump.json]:
```suggestion
  "name": "Test surv\u00eby2",
```

File: survey/tests/testdump.json (hunk 3 above)
Comment (giliam) [reply survey/tests/testdump.json]: Fixed!

**Pull request merged**